core/src/com/unciv/logic/civilization/managers/GreatPersonManager.kt

```diff
@@ -49,10 +58,15 @@ class GreatPersonManager : IsPartOfGameInfoSerialization {
         }
 
         for ((key, value) in greatPersonPointsCounter) {
-            val requiredPoints = getPointsRequiredForGreatPerson()
+            val requiredPoints = getPointsRequiredForGreatPerson(key)
             if (value >= requiredPoints) {
                 greatPersonPointsCounter.add(key, -requiredPoints)
-                pointsForNextGreatPerson *= 2
+                for (unique in civInfo.getEquivalentUnit(key).getMatchingUniques(UniqueType.GreatType, stateForConditionals = StateForConditionals.IgnoreConditionals))
+                {
+                    pointsForNextGreatPerson[unique.params[0]] *= 2
+                    return key
+                }
+                pointsForNextGreatPerson[""] *= 2
```

Comment:
1 gp in several pools would be possible in the current arrangement but wouldn't update correctly. If that's what's wanted, then I'd have to switch it to return after checking all of the pools it may be a part of and keeping an extra variable to track if it found at all. And given the previous comment saying you don't like it being a loop, that would be unavoidable if we allow 1 gp to be in multiple pools

As for relying on the default, I'm not sure I understand your question. The use of the default pool is just a fallback case in case nothing is found. This is to keep it working with current mods seamlessly. If the default pool was to be explicitly used, it actually wouldn't change anything as fall as I'm aware

Reply:
I'll have to double check tomorrow, but I'm pretty sure as it's set up, it if finds a case of the unique, it will only increase the pool it found, not the default, due to the use of the return. Still, your solution is much nicer, so it's definitely the better way to go unless Yairm would rather we allow 1 gp to be in multiple pools